### ISSUE TITLE
no-unused-variable false positive with references in JSX

### ISSUE THREAD

**Issue by adidahiya** [palantir/tslint#558]
code sample reported by @pepaar:

```ts
/// <reference path="react.d.ts" />
/// <reference path="react-jsx.d.ts" />

import React = require("react");

interface IProps {
    content: string;
}

class MyComponent extends React.Component<IProps, {}> {
    render() {
        this.alertMe();
        return 
            <div>
                {this.props.content}
                <button onClick={() => this.alertMe()}>AlertMe</button>
            </div>;
    }

    private alertMe() {
        let yo = this.props.content + " YO";
        alert(yo);
    }
}

React.render(<MyComponent content="Hello World"/>, document.body);
```

`languageService.getDocumentHighlights` doesn't return the right thing for `alertMe` in this case.

seems to be caused by a compiler bug: https://github.com/Microsoft/TypeScript/issues/4175

...but @pepaar also reported a linter crash, which should be fixed on our end:

```
C:\Users\pepaar\AppData\Roaming\npm\node_modules\tslint\build\rules\noUnusedVari
ableRule.js:161
        if (highlights[0].highlightSpans.length <= 1) {
                      ^
TypeError: Cannot read property '0' of undefined
    at NoUnusedVariablesWalker.validateReferencesForVariable (C:\Users\pepaar\Ap
pData\Roaming\npm\node_modules\tslint\build\rules\noUnusedVariableRule.js:161:23
)
...
```

__UPDATE__: the linter crash has been fixed by #563
